File: Data_Structures/Arrays/longest_subarray.c
//Given an array of integers, find the longest increasing subarray (contiguous elements) and return its length

#include <stdio.h>

int longestIncreasingSubarray(int arr[], int size) {
    int maxLength = 1;
    int currentLength = 1;

    for (int i = 1; i < size; i++) {
        if (arr[i] > arr[i - 1]) {
            currentLength++;
        } else {
            if (currentLength > maxLength) {
                maxLength = currentLength;
            }
            currentLength = 1;
        }
    }

    if (currentLength > maxLength) {
        maxLength = currentLength;
    }

    return maxLength;
}

int main() {
    int arr[] = {1, 2, 3, 2, 5, 7, 8, 9, 2, 4};
    int size = sizeof(arr) / sizeof(arr[0]);
    int length = longestIncreasingSubarray(arr, size);

    printf("Longest increasing subarray length: %d\n\n", length);

    return 0;
}

File: Data_Structures/Arrays/longest_subarray.py
# Given an array of integers, find the longest increasing subarray (contiguous elements) and return its length

def longest_increasing_subarray(arr):
    maxLength = 1
    currentLength = 1

    for i in range(1, len(arr)):
        if arr[i] > arr[i - 1]:
            currentLength += 1
        else:
            if currentLength > maxLength:
                maxLength = currentLength
            currentLength = 1

    if currentLength > maxLength:
        maxLength = currentLength

    return maxLength

arr = [1, 2, 3, 2, 5, 7, 8, 9, 2, 4]
length = longest_increasing_subarray(arr)
print("Longest increasing subarray length:", length)
print("\n")
